internal/downloader/downloader.go

```diff
@@ -66,31 +60,11 @@ func Download(ctx context.Context, destDir string, sourceUrl string, showMsg boo
 	}
 
 	dl := func(ctx context.Context, sourceUrl, destDir string) (metadata.Metadata, error) {
-		// conftest's Download function leverages oras under the hood to fetch from OCI. It uses the
-		// global oras client and sets the user agent to "conftest". This is not a thread safe
-		// operation. Here we get around this limitation by ensuring a single download happens at a
-		// time.
-		dlMutex.Lock()
-		defer dlMutex.Unlock()
-		return nil, downloader.Download(ctx, destDir, []string{sourceUrl})
-	}
-
-	if utils.UseGoGather() {
-		dl = func(ctx context.Context, sourceUrl, destDir string) (metadata.Metadata, error) {
-			m, err := gatherFunc(ctx, sourceUrl, destDir)
-			if err != nil {
-				log.Debug("Download failed!")
-			}
-			return m, err
-		}
-	} else if d, ok := ctx.Value(downloadImplKey).(downloadImpl); ok {
-		dl = func(ctx context.Context, sourceUrl, destDir string) (metadata.Metadata, error) {
-			err := d.Download(ctx, destDir, []string{sourceUrl})
-			if err != nil {
-				log.Debug("Download failed!")
-			}
-			return nil, err
+		m, err := gatherFunc(ctx, sourceUrl, destDir)
+		if err != nil {
+			log.Debug("Download failed!")
 		}
+		return m, err
 	}
 
 	m, err := dl(ctx, sourceUrl, destDir)
```

Comment:
We probably don't need to create a `dl` wrapper function and can just call `gatherFunc` directly here.

Reply:
+1, wdyt Rob.